### ISSUE TITLE
Kibana 4.x plugin installation via Url not working

### ISSUE THREAD

**Issue by lucabelluccini** [elastic/puppet-kibana#7]
## Bug Report

* Module version:  6d8dd95
* Puppet version: 4.7.0
* OS and version: Centos 6.8

### Bug description

Installation of plugin via url on Kibana 4 doesn't work, but status reported OK

Expected behavior:

Plugin installed or explicit error during puppet agent run.

Observed behavior:

Resource created, but no plugin installed.
Organization is REQUIRED but if url is provided it would not be necessary.
Plugin resource reported as created but nothing happened. Should report:

```
/opt/kibana/bin/kibana plugin --install https://xxxxxxxxxxxxxxxx/marvel-2.4.4.tar.gz
Invalid install option. Please use the format <org>/<plugin>/<version>.
```

Example manifest:

custom_module.pp
```
  $kibana_plugins = hiera_hash('profile::kibana::plugins', {})
  create_resources('kibana_plugin', $kibana_plugins)

```

hieralevel.yaml
```
profile::kibana::plugins:
  'marvel':
    url: 'https://xxxxxxxxxxxxxxx/marvel-2.4.4.tar.gz'
    organization: 'elasticsearch' # Tried with and without it
    version: '2.4.4' # Tried with and without it
```

Puppet run logs:

```
Debug: Executing: '/sbin/service kibana status'
Debug: Executing: '/sbin/chkconfig kibana'
Debug: Prefetching kibana resources for kibana_plugin
Notice: /Stage[main]/Profile::Kibana/Kibana_plugin[marvel]/ensure: created
Debug: Executing with uid=kibana gid=kibana: '/opt/kibana/bin/kibana plugin --install https://xxxxxxxxxxxxx/marvel-2.4.4.tar.gz'
Debug: Kibana_plugin[marvel](provider=kibana): 
Debug: /Stage[main]/Profile::Kibana/Kibana_plugin[marvel]: The container Class[Profile::Kibana] will propagate my refresh event
```

## Feature Request

Allow installation via url and name also for Kibana 4.6

### Feature Description

Installation via url is available executing ```kibana plugin --install <name> --url <url>``` in Kibana 4.6 (even before).

**Comment by tylerjl** [comment]
Hey @lucabelluccini! Yeah, I can see the bug here. Based upon what I'm seeing, let me know if this sounds right:

* If a `url` is specified for a `kibana_plugin`, the install command should go from:
  `/opt/kibana/bin/kibana plugin --install https://xxxxxxxxxxxxxxxx/marvel-2.4.4.tar.gz`
  to:
  `/opt/kibana/bin/kibana plugin --install marvel --url https://xxxxxxxxxxxxxxxx/marvel-2.4.4.tar.gz`

That's also weird a nonzero exit didn't raise a Puppet error, I can look into that as well.

**Comment by lucabelluccini** [comment]
Hello!

Yes, you're right.

For the command, if url is specified, other params (except $title/$namevar) can be ignored and command should be tweaked.
You can take a look [here](https://github.com/LucaBelluccini1A/jlambert121-kibana/blob/master/manifests/plugin.pp#L29)

For the exit status, I think it is not checked.

**Comment by tylerjl** [comment]
The wrong --url behavior has been fixed in #8 and I've also fixed nonzero exit codes being eaten in a806f6b6 - I'll cut a release after these fixes as the --url one is pretty important. Thanks for raising the issue!
